packages/pyright-internal/src/analyzer/binder.ts

```diff
@@ -386,6 +387,19 @@ export class Binder extends ParseTreeWalker {
             }
         }
 
+        // Add warning diagnostic for V2 micro:bit modules.
+        maybeAddMicrobitVersionWarningBinderWrapper(importResult.importName, () =>
+            this._addDiagnostic(
+                this._fileInfo.diagnosticRuleSet.reportMicrobitV2ApiUse,
+                DiagnosticRule.reportMicrobitV2ApiUse,
+                Localizer.Diagnostic.microbitV2ModuleUse().format({
+                    moduleName: importResult.importName,
+                    device,
+                }),
+                node
+            )
+        );
```

Comment:
This is setup in this way so we can track micro:bit specific diagnostics in one file, `microbitUtils.ts`. Useful if we only want to warn on the first use of a V2 API call in the future.